packages/tradershub/src/components/UIProvider/UIProvider.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { createContext, useCallback, useContext, useMemo, useState } from 'react';
+
+type TUIState = {
+    // Add other UI states here
+    accountType?: string;
+    region?: string;
+};
+
+type TUIContext = {
+    getUIState: <T extends keyof TUIState>(key: T) => TUIState[T];
+    setUIState: <T extends keyof TUIState>(key: T, value: TUIState[T]) => void;
+};
+
+const UIContext = createContext<TUIContext | null>(null);
+
+export const useUIContext = () => {
+    const context = useContext(UIContext);
+
+    if (!context) {
+        throw new Error('useUIContext must be used within a UIProvider');
+    }
+
+    return context;
+};
+
+export const UIProvider = ({ children }: { children: React.ReactNode }) => {
+    const [uiState, setUIState] = useState<TUIState>({});
```

Comment:
```suggestion
    const [UIState, setUIState] = useState<TUIState>({});
```

Reply:
Sonarcloud will complain if i do this

![Screenshot 2024-01-17 at 8 56 38 AM](https://github.com/binary-com/deriv-app/assets/104053934/a5c8fc72-8c63-4752-b771-3d21dc4c6640)